src/create-context.js

```diff
@@ -21,18 +21,21 @@ export function createContext(defaultValue) {
 
 				this.shouldComponentUpdate = _props => {
 					if (this.props.value !== _props.value) {
-						subs.some(c => {
-							c.context = _props.value;
-							enqueueRender(c);
+						subs.some(payload => {
+							payload.c.context = _props.value;
+							if (!payload.u || payload.u(_props.value, this.props.value)) {
+								enqueueRender(payload.c);
+							}
 						});
 					}
 				};
 
-				this.sub = c => {
-					subs.push(c);
+				this.sub = (c, shouldUpdate) => {
+					const entry = { c, u: shouldUpdate };
```

Comment:
Wondering if we want to allow for this function to be updated, my gut sais no. If we want to we could do it with a mutable ref?

Reply:
wouldn't that require unsubscribing and resubscribing right now? Only way to do it without that would be to pass an object, which is sortof a strange API signature.

Reply:
Agreed, unless we do like `x.sub({ current: shouldUpdate })` in Consumer and useContext